lib/src/results.dart

```diff
@@ -135,6 +135,85 @@ extension RealmResultsOfObject<T extends RealmObjectBase> on RealmResults<T> {
   }
 }
 
+class _SubscribedRealmResult<T extends RealmObject> extends RealmResults<T> {
+  final String? subscriptionName;
+
+  _SubscribedRealmResult._(RealmResults<RealmObject> results, {this.subscriptionName})
+      : super._(
+          results.handle,
+          results.realm,
+          results.metadata,
+        );
+}
+
+extension RealmResultsOfRealmObject<T extends RealmObject> on RealmResults<T> {
+  /// Adds this [RealmResults] query to the set of active subscriptions.
+  /// The query will be joined via an OR statement with any existing queries for the same type.
+  ///
+  /// If a [name] is given this allows you to later refer to the subscription by name,
+  /// e.g. when calling [MutableSubscriptionSet.removeByName].
+  ///
+  /// If [update] is specified to `true`, then any existing query
+  /// with the same name will be replaced.
+  /// Otherwise a [RealmException] is thrown, in case of duplicates.
+  ///
+  /// [WaitForSyncMode] specifies how to wait or not wait for subscribed objects to be downloaded.
+  /// The default value is [WaitForSyncMode.onCreation].
+  ///
+  /// The [timeout] is the maximum duration to wait for objects to be downloaded.
+  /// If the time exceeds this limit, the [RealmResults] is returned and the download
+  /// continues in the background.
+  ///
+  /// {@category Sync}
+  Future<RealmResults<T>> subscribe({
+    String? name,
+    WaitForSyncMode waitForSyncMode = WaitForSyncMode.onCreation,
+    Duration? timeout,
+    bool update = false,
+  }) async {
+    bool subscriptionExists = realm.subscriptions.find(this) != null;
+    realm.subscriptions.update((mutableSubscriptions) {
+      mutableSubscriptions.add(this, name: name, update: update);
+    });
+
+    if (waitForSyncMode == WaitForSyncMode.always || (waitForSyncMode == WaitForSyncMode.onCreation && !subscriptionExists)) {
+      Future<void> waitForDownload() => realm.subscriptions.waitForSynchronization().then((value) => realm.syncSession.waitForDownload());
+      if (timeout == null) {
+        await waitForDownload();
+      } else {
+        await waitForDownload().timeout(timeout, onTimeout: () {});
```

Comment:
I think we should mention in the doc comment that a timeout is silent, ie. won't raise an exception. That is not common behaviour in Dart.